### PR TITLE
feat: Type hinting for for nodejs hello-world template

### PR DESCRIPTION
**Pull request by iRoachie**
*Description of changes:*
Adds type hints and autocomplete for context, and event parameters on the lambdaHandler.

Knowing what values exist in these parameters can be daunting, and usually require going to the links provided in the jsdoc. Having types here reduces that friction and keeps the developer in the editor.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/hello-world/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "compilerOptions": {
```
Comment (jfuss) [inline samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/hello-world/tsconfig.json]: Forgive me, as I am not super familiar with Nodejs. It seems you are converting this example from Nodejs to Typescript? I can see the benefit but we want to keep the examples usable with Nodejs without the need to convert Typescript to Nodejs.

Comment (iRoachie) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/hello-world/tsconfig.json]: Nope, the project stays as NodeJs. What's happening here is that we're using typescript power the types. So you write normal javascript, but get type-hinting and completion from typescript

Comment (aahung) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/hello-world/tsconfig.json:2]: I think if we want to do this, it is better to do it for all templates in https://github.com/aws/aws-sam-cli-app-templates

Comment (iRoachie) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/hello-world/tsconfig.json:2]: I'm willing to add the other node templates to this PR if you all are interested still

**Comment by aahung** [comment]
Close in favor of PRs in https://github.com/aws/aws-sam-cli-app-templates

We also have a PR adding typescript template https://github.com/aws/aws-sam-cli-app-templates/pull/77

If you are still interested, can you create PRs there?

File: samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/hello-world/tsconfig.json (hunk 1 above)
Comment (iRoachie) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/hello-world/tsconfig.json:2]: Hey @aahung when you say all templates, you're referring to all the templates in `nodejs10.x`, `nodejs12.x`, `nodejs14.x`, including docker ones?

Comment (aahung) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/hello-world/tsconfig.json:2]: Yeah, I think it is good for consistency across templates.

Comment (iRoachie) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/hello-world/tsconfig.json:2]: One more thing @aahung. Some templates have more than one lambda in them so I personally don't think it makes sense to add the dependency to each function's package.json. At work, we do a top-level package.json. However it means we'd need to update the documentation as well.

Here's an example https://github.com/iRoachie/aws-sam-cli-app-templates/commit/9c303959532bce8c325b02794b489e789545efb6.

Notice the directory tree:

```
{{cookiecutter.project_name}}
 ┣ events
 ┃ ┗ event.json
 ┣ hello-world
 ┃ ┣ tests
 ┃ ┃ ┗ unit
 ┃ ┃ ┃ ┗ test-handler.js
 ┃ ┣ .npmignore
 ┃ ┣ app.js
 ┃ ┗ package.json
 ┣ .gitignore
 ┣ README.md
 ┣ package.json // this is where @types/aws-lambda will be installed
 ┗ template.yaml
```


Let me know what you think about this approach.

**Pull request closed without merging**